eclair-core/src/main/scala/fr/acinq/eclair/db/sqlite/SqliteAuditDb.scala

```diff
@@ -278,17 +278,18 @@ class SqliteAuditDb(sqlite: Connection) extends AuditDb with Logging {
           rs.getLong("timestamp"))
         relayedByHash = relayedByHash + (paymentHash -> (relayedByHash.getOrElse(paymentHash, Nil) :+ part))
       }
-      relayedByHash.map {
-        case (paymentHash, parts) => parts.head.relayType match {
-          case "channel" => parts.foldLeft(ChannelPaymentRelayed(0 msat, 0 msat, paymentHash, ByteVector32.Zeroes, ByteVector32.Zeroes, parts.head.timestamp)) {
-            case (e, part) if part.direction == "IN" => e.copy(amountIn = part.amount, fromChannelId = part.channelId)
-            case (e, part) if part.direction == "OUT" => e.copy(amountOut = part.amount, toChannelId = part.channelId)
+      relayedByHash.flatMap {
+        case (paymentHash, parts) =>
+          val timestamp = parts.head.timestamp
+          // We may have been routing multiple payments for the same payment_hash (MPP) in both cases (trampoline and channel).
+          val incoming = parts.filter(_.direction == "IN").map(p => PaymentRelayed.Part(p.amount, p.channelId))
+          val outgoing = parts.filter(_.direction == "OUT").map(p => PaymentRelayed.Part(p.amount, p.channelId))
+          parts.head.relayType match {
+            case "channel" => incoming.zip(outgoing).map {
```

Comment:
```suggestion
            case "channel" => incoming.zip(outgoing).map { // NB: we may link the wrong in-out parts, but the overall sum will be correct
```

we could sort both lists to reduce the chance of this happening

Reply:
Improved that a bit in `16dd4ff`